examples/Cloudy/Cloudy_example.jl

```diff
@@ -216,42 +253,40 @@ mcmc = MCMC.MCMCObj(yt_sample, Γy, priors, new_step, u0, max_iter,
                     mcmc_alg, burnin, svdflag=true)
 MCMC.sample_posterior!(mcmc, gpobj, max_iter)
 
-posterior = MCMC.get_posterior(mcmc)      
+posterior = MCMC.get_posterior(mcmc)
 
-post_mean = mean(posterior, dims=1)
-post_cov = cov(posterior, dims=1)
+post_mean = get_mean(posterior)
+post_cov = get_cov(posterior)
 println("post_mean")
 println(post_mean)
 println("post_cov")
 println(post_cov)
-println("D util")
-println(det(inv(post_cov)))
-println(" ")
 
 # Plot the posteriors together with the priors and the true parameter values
-true_values = [log(N0_true) log(θ_true) log(k_true)]
-n_params = length(true_values)
+n_params = length(names)
 
 for idx in 1:n_params
     if idx == 1
-        param = "N0"
-        xs = collect(4.5:0.01:6.5)
+        xs = collect(range(-2.0, stop=2.0, length=1000))
     elseif idx == 2
-        param = "Theta"
-        xs = collect(-1.0:0.01:2.5)
+        xs = collect(range(-2.0, stop=2.0, length=1000))
     elseif idx == 3
-        param = "k"
-        xs = collect(-4.0:0.01:1.0)
+        xs = collect(range(-2.0, stop=2.0, length=1000))
     else
         throw("not implemented")
     end
 
-    label = "true " * param
-    histogram(posterior[:, idx], bins=100, normed=true, fill=:slategray, 
+    label = "true " * names[idx]
+    println("size of samples to be plotted in histogram")
+    println(size(posterior.distributions[idx].distribution_samples))
+    ys = dropdims(posterior.distributions[idx].distribution_samples, dims=1)
```

Comment:
Could you use the proper getter method here? e.g `get_n_samples(posterior)` returns a dictionary for each distribution, if they are of `Samples` type it will have the number of samples.

Reply:
These were debugging comments that have been removed.